lib/suma/admin_api/common_endpoints.rb

```diff
@@ -20,6 +41,8 @@ def update_model(m, orig_params, process_params: nil, save: true)
       to_many_assocs_and_args = []
       to_one_assocs_and_params = []
       fk_attrs = {}
+      # handle roles separately
+      roles = params.delete(:roles)
```

Comment:
You need to do this in the endpoint code, before you call `update_model`. We shouldn't special case parameter names here.